pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -121,17 +120,17 @@ func (dv *openShiftClusterDynamicValidator) validateServicePrincipalProfile(ctx
 	defer cancel()
 
 	// get a token, retrying only on AADSTS700016 errors (slow AAD propagation).
-	// As we don't do `err = wait.PollImmediateUntil()`, we won't ever see a
-	// "timed out waiting for the condition" error, but instead will always see
-	// the last error returned from token.EnsureFresh().
-	wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+	err = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
 		err = token.EnsureFresh()
```

Comment:
[optional] should the inner err be writing to the outer variable? i have seen in places
```
innerErr := foo()
...
```
I don't think there is a logic difference even when err has the AADSTS (err will temporarily have something in it)